modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java

```diff
@@ -907,18 +909,27 @@ private String getTypeString(Schema p, String prefix, String suffix) {
         if (")".equals(suffix)) {
             fullSuffix = "," + suffix;
         }
-        if (ModelUtils.isAnyTypeSchema(p)) {
+        if (StringUtils.isNotEmpty(p.get$ref())) {
+            // The input schema is a reference. If the resolved schema is
+            // a composed schema, convert the name to a Python class.
+            Schema s = ModelUtils.getReferencedSchema(this.openAPI, p);
+            if (s instanceof ComposedSchema) {
+                String modelName = ModelUtils.getSimpleRef(p.get$ref());
+                return prefix + toModelName(modelName) + fullSuffix;
```

Comment:
@spacether , here the type may end up being used in the `codegenModel.additionalPropertiesType` property, which is set by `addAdditionPropertiesToCodeGenModel()`. I need to add that type to the imports.

Reply:
Fixed in f57338d2110e0694dec7c0e07c9966c914c74359